File: roomDemo_kt/app/src/main/java/edu/cosc4730/roomdemo_kt/AppDatabase.java
package edu.cosc4730.roomdemo_kt;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

/**
 *  This is very simple code.  This is the "database" class and provides a bridge via SocreDoa method
 *  to the data itself. This is the simplest version of the RoomDatabase class.
 *
 *  note this is java code, since the Kotlin code won't always work at runtime.  likely something
 *  to do with the static piece.
 */
@Database(entities = {Score.class,}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase db;

    public static final String DATABASE_NAME = "database-name.db";

    public abstract ScoreDao ScoreDao();

    public static AppDatabase getInstance(final Context context) {
        if (db == null) {
            db = Room.databaseBuilder(context, AppDatabase.class, DATABASE_NAME).build();
        }
        return db;
    }

}

File: roomDemo_kt/app/src/main/java/edu/cosc4730/roomdemo_kt/AppDatabase.kt
package edu.cosc4730.roomdemo_kt

import android.content.Context
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase

/**
 * This is very simple code.  This is the "database" class and provides a bridge via SocreDoa method
 * to the data itself. This is the simplest version of the RoomDatabase class.
 *
 * Room implementation code is generated at build time by KSP.
 * The Gradle Kotlin DSL files must keep the Room compiler on the `ksp` configuration
 * or AppDatabase_Impl will not be generated.
 */
@Database(entities = [Score::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun ScoreDao(): ScoreDao

    companion object {
        private var db: AppDatabase? = null

        const val DATABASE_NAME: String = "database-name.db"

        @JvmStatic
        fun getInstance(context: Context): AppDatabase {
            if (db == null) {
                db = databaseBuilder(context, AppDatabase::class.java, DATABASE_NAME).build()
            }
            return db!!
        }
    }
}
